modules/store/src/selector.ts

```diff
@@ -17,13 +17,19 @@ export type ComparatorFn = (a: any, b: any) => boolean;
 
 export type DefaultProjectorFn<T> = (...args: any[]) => T;
 
+type SelectorProjectorFn<ProjectorFn> = ProjectorFn extends (
+  ...args: infer ProjectorArgs
+) => infer ProjectorResult
+  ? (...args: ProjectorArgs) => ProjectorResult
+  : ProjectorFn;
+
```

Comment:
Do we really need the extra type here? It seems to work without `SelectorProjectorFn`.

```suggestion

```

Reply:
Yeah, this is unneeded now. This was an artifact of attempting to deal with types that may not have been functions passed into the generic, but should have been cleaned up.